### PR TITLE
Add more of the Instant Launch API

### PR DESCRIPTION
**Pull request by johnworth**
All of this will be needed for administrating instant launches in Sonora. Adds the ability to manage the instant-launch to file mapping, along with the instant launch objects themselves.

### CODE REVIEW

File: src/terrain/routes/schemas/instantlaunches.clj [hunk 1]
```diff
@@ -3,11 +3,36 @@
...
+   which determine which files can be used with a particular instant launch")
+
 (defschema InstantLaunch
-  {:id              (describe UUID "The UUID of the instant launch")
-   :quick_launch_id (describe UUID "The UUID for the quick launch used for the instant launch")
-   :added_by        (describe String "The username of the user that created the instant launch")
-   :added_on        (describe String "The date and time when the instant launch was created")})
+  {(optional-key :id)              (describe UUID "The UUID of the instant launch")
+   :quick_launch_id                (describe UUID "The UUID for the quick launch used for the instant launch")
+   (optional-key :added_by)        (describe String "The username of the user that created the instant launch")
```
Comment (slr71) [inline src/terrain/routes/schemas/instantlaunches.clj:31]: This is minor, but `user who created the instant launch` is probably more correct.

**Review by slr71 — APPROVED** [review]
I had a couple of comments, but there's nothing that absolutely needs to be addressed. :shipit: whenever you're ready.

File: src/terrain/routes/schemas/instantlaunches.clj (hunk 1 above)
Comment (johnworth) [reply src/terrain/routes/schemas/instantlaunches.clj]: Oops, yup. Should be fixed now.

File: src/terrain/clients/app_exposer.clj [hunk 2]
```diff
@@ -94,4 +94,63 @@
...
+  (-> (app-exposer-url ["instantlaunches" id] {} :no-user true)
+      (client/get {:as :json})
+      (:body)))
+
+(defn add-instant-launch
+  [username il]
+  (-> (app-exposer-url ["instantlaunches/"] {} :no-user true)
+      (client/put {:content-type :json
+                   :as           :json
+                   :form-params  (update il :added_by #(if (not %1) username %1))})
```
[the remark this answers is not in the document]
Comment (johnworth) [reply src/terrain/clients/app_exposer.clj]: Thanks, should be fixed now.

**Pull request merged**